### DISCUSSION TITLE
Can `NeighborSampler` support weights ?

### DISCUSSION THREAD

**Discussion by xiachenrui** [pyg-team/pytorch_geometric#7295] [Q&A] [answered]
Hi, all:

Can we sample the graph with (edge) weight  when using [`NeighborSampler`](https://pytorch-geometric.readthedocs.io/en/latest/modules/sampler.html#torch_geometric.sampler.NeighborSampler) ? This strategy is avaliable in DGL (https://docs.dgl.ai/en/0.8.x/generated/dgl.sampling.sample_neighbors.html#dgl.sampling.sample_neighbors)

Best !
Chen-Rui

**Comment by wsad1** [comment, marked as answer]
`NeighborSampler` has been depreciated in favour of `NeighborLoader`. And `NeighborLoader` currently doesn't support defining a weight/prob for each edge which can be used while sampling.
But this a useful feature to add @mananshah99 , @rusty1s.

**Reply by xiachenrui** [reply]
Thanks for your kind reply !

**Reply by rusty1s** [reply]
Yes, we have this on our agenda with @pyg-team/intel-team :)
